vortex-layout/src/data.rs

```diff
@@ -75,18 +75,17 @@ impl LayoutData {
     }
 
     /// Create a new viewed layout from a flatbuffer root message.
-    pub fn try_new_viewed(
+    ///
+    /// SAFETY:
+    ///
+    /// Assumes that flatbuffer has been previously validated
+    pub fn try_new_viewed_unchecked(
```

Comment:
* If it's unchecked, why does it also return a result?
* Shouldn't it be marked as unsafe?
* Where is the previous validation? How can this function's comment know about previous validation? I think that should be the comment at the call site, this comment should explain what preconditions are required.